modules/javafx.graphics/src/main/native-glass/monocle/egl/eglBridge.c

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+#include "com_sun_glass_ui_monocle_EGLAcceleratedScreen.h"
+#include "Monocle.h"
+#include "egl_ext.h"
+
+JNIEXPORT jlong JNICALL Java_com_sun_glass_ui_monocle_EGLAcceleratedScreen_nPlatformGetNativeWindow
+    (JNIEnv *env, jobject obj, jstring cardId) {
+    const char *ccid = (*env)->GetStringUTFChars(env, cardId, NULL);
+    long answer = getNativeWindowHandle(ccid);
+    (*env)->ReleaseStringUTFChars(env, cardId, ccid);
+    return (jlong)answer;
+}
+
+JNIEXPORT jlong JNICALL Java_com_sun_glass_ui_monocle_EGLAcceleratedScreen_nGetEglDisplayHandle
+    (JNIEnv *env, jobject obj) {
+    long answer = getEGLDisplayHandle();
+    return (jlong)answer;
+}
+
+JNIEXPORT jboolean JNICALL Java_com_sun_glass_ui_monocle_EGLAcceleratedScreen_nEglInitialize
+    (JNIEnv *env, jclass clazz, jlong eglDisplay) {
+    jboolean answer = doEglInitialize(asPtr(eglDisplay));
+    return answer;
+}
+
+JNIEXPORT jboolean JNICALL Java_com_sun_glass_ui_monocle_EGLAcceleratedScreen_nEglBindApi
+    (JNIEnv *env, jclass clazz, jint api) {
+    jboolean answer = doEglBindApi((int)api);
+    return answer;
+}
+
+JNIEXPORT jlong JNICALL Java_com_sun_glass_ui_monocle_EGLAcceleratedScreen_nEglChooseConfig
+    (JNIEnv *env, jclass clazz, jlong eglDisplay, jintArray attribs) {
+    jint *attrArray = (*env)->GetIntArrayElements(env, attribs, JNI_FALSE);
```

Comment:
Should this check the return value in case it fails (returns NULL)?

Reply:
I checked other places in the code, and calls to env->GetIntArrayElements are typically not checked.
However, if something fails in that call, it must be something really bad that happened. I added a fprintf to stderr, and throw an IllegalArgumentException in the java code (as there is something wrong with the attribs)